rosjava/src/main/java/org/ros/internal/node/service/DefaultServiceClient.java

```diff
@@ -119,11 +120,8 @@ public DefaultServiceClient(final GraphName nodeName,
     connectionHeader.addField(ConnectionHeaderFields.PERSISTENT, "1");
     connectionHeader.merge(serviceDeclaration.toConnectionHeader());
     tcpClientManager = new TcpClientManager(executorService);
-    final ServiceClientHandshakeHandler<T, S> serviceClientHandshakeHandler =
-        new ServiceClientHandshakeHandler<T, S>(connectionHeader, responseListeners, deserializer,
+    serviceClientHandshakeHandler = new ServiceClientHandshakeHandler<T, S>(connectionHeader, responseListeners, deserializer,
             executorService);
-    handshakeLatch = new HandshakeLatch();
-    serviceClientHandshakeHandler.addListener(handshakeLatch);
```

Comment:
Each time this line is touched, a thread is leaked.
A new thread is created to wait for the coming handshake response, but they are never released no matter the handshake is successful or not. As a result, there's always be a new zombie thread after each service client creation.

In ordinary cases, there are only a small number of service clients, so this problem is not serious.
As we did service clients cache and recreation to simulate non-persist service calling, it explodes when there are many failure and many clients are created.